src/dnf-plugins/product-id.py

```diff
@@ -108,30 +119,84 @@ def get_enabled(self):
                       self.meta_data_errors)
         return lst
 
+    @staticmethod
+    def _get_available():
+        """Try to get list of available packages"""
+
+        # We create own base object to get fresh list of installed packages
+        # and to get list of available packages after installing/removing,
+        # because base object from dnf.Plugin provides incomplete and obsolete data.
+
+        # FIXME: It is not safe to use two base objects in transaction hook.
+        # Try to remove it, when dnf support getting list of available
+        # packages during "dnf remove".
+        with dnf.base.Base() as base:
+            base.read_all_repos()
+            base.fill_sack(load_system_repo=True, load_available_repos=True)
+            available = base.sack.query().available()
```

Comment:
available represents a query with sack from object (base) that is going to die. Returning of such a object is quite risky. Additionally loading of second sack with available packages could result:
1. another copy of cash is created
2. Data could be different from data available to running command (due to different setting)
3. It can overwrite solv files of running command - crash
4. Huge performance problem
Only what you can request is sack with system repo base.fill_sack(load_system_repo='auto', load_available_repos=False).
Please try to convince yourself, that you don't need additional resources during dnf run. If command doesn't needed you cannot modify staff on the system that requires available repository.